src/encoded/static/components/matrix_reference_epigenome.js

```diff
@@ -647,43 +756,80 @@ class MatrixPresentation extends React.Component {
             };
         }
 
+        const rowCount = matrix => matrix.y['biosample_ontology.classification'].buckets.reduce((currentTotalCount, termName) => currentTotalCount + termName['biosample_ontology.term_name'].buckets.length, 0);
+
         return (
-            <div className="matrix__presentation">
-                <div className={`matrix__label matrix__label--horz${!scrolledRight ? ' horz-scroll' : ''}`}>
-                    <span>{context.matrix.x.label}</span>
-                    {svgIcon('largeArrow')}
+            <React.Fragment>
+                <div className="view-controls-container">
+                    <ViewControls results={this.props.context} alternativeNames={['Search list', 'Tabular report', 'Summary matrix']} />
+                    <BatchDownloadControls results={context} />
                 </div>
-                <div className="matrix__presentation-content">
-                    <div className="matrix__label matrix__label--vert"><div>{svgIcon('largeArrow')}{context.matrix.y.label}</div></div>
-                    <TabPanel tabs={organismTabs} selectedTab={this.initialSelectedTab} handleTabClick={this.handleTabClick} tabPanelCss="matrix__data-wrapper">
-                        {!this.state.organismChooserVisible ?
-                            <div className="matrix__data" onScroll={this.handleOnScroll} ref={(element) => { this.scrollElement = element; }}>
-                                {matrixConfig ?
-                                    <DataTable tableData={matrixConfig} />
-                                : null}
+                <div className="summary-controls">
+                    <div className="matrix-header__filter-controls">
+                        <SearchFilter context={context} />
+                    </div>
+                    <div className="results-count">Showing <b className="bold-total">{rowCount(context.matrix)}</b> result{rowCount(context.matrix) > 1 ? 's' : ''}.</div>
+                </div>
+                <TabPanel tabs={organismTabs} selectedTab={this.initialSelectedTab} handleTabClick={this.handleTabClick} tabPanelCss="matrix__data-wrapper">
+                    {(this.initialSelectedTab === 'Homo sapiens') ?
```

Comment:
If you go with my idea of swapping they keys and values of organismTerms, you can use it here (organismTerms['human'])

Reply:
Just simplifying this.